shell-switcher.el

```diff
@@ -268,7 +268,7 @@ key is pressed, calls `sswitcher--switch-partially'."
 	 `(lambda () (interactive)
 	    (shell-switcher--switch-partially)
 	    (message ,message)))
-       map) t)
+       map) t (lambda () (switch-to-buffer (current-buffer))))
```

Comment:
The new sexp here is executed once they press something other than '.  It switches to itself, but in a way that places the current buffer on the top of the buffer stack that is returned by (buffer-list).  When there is no third parameter passed to switch-to-buffer, it will place the given buffer on top of the stack.  Other emacs packages may rely on the fact that the current working buffer is on top of the stack.  I can work on a feature test when I get some time.